### DISCUSSION TITLE
Incomplete LSP completion with company

### DISCUSSION THREAD

**Discussion by azegas** [emacs-lsp/lsp-mode#3781] [Q&A] [answered]
Hello,

I had a fully working LSP configuration before reinstalling my ubuntu system. I am now getting not so full LSP experience, because autocompletion does not work as I was used to. 

It seems that it does not understand how to complete properly. Current completion reminds me of yasnippet snippet...

Plese let me know what other information about my system could be helpful to debug this issue.

GNU Emacs 28.1

Here is my lsp config - https://github.com/arvydasg/emacs/blob/master/recipes/lsp-mode.el 

**LSP-Doctor output:**

![image](https://user-images.githubusercontent.com/78803192/197594596-be316f63-2502-4811-9972-771bfa702462.png)

**Gif of a problem:**

![Peek 2022-10-24 20-51](https://user-images.githubusercontent.com/78803192/197592742-f685fa39-7822-4f3f-8527-845d9bd37d65.gif)

**Comment by yyoncho** [comment, marked as answer]
Can you make sure that yasnippet is installed and that you have yasnippet-mode enabled?

**Reply by yyoncho** [reply]
I can confirm that it works fine with M-x lsp-start-plain and most likely you haven't loaded your https://github.com/arvydasg/emacs/blob/master/recipes/yasnippet.el file. Generally, it is lsp-mode bug(we should disable snippets when yas is not enabled/on). It is recent regression.

**Reply by azegas** [reply]
I am loading yasnippet globally in every mode.

I thought perhaps not all the yasnippet snippets got installed, so I used [this](https://github.com/AndreaCrotti/yasnippet-snippets) as a reference and did:

`M-x package-install yasnippet-snippets`
and
`sudo apt install elpa-yasnippet-snippets`

Still the same.. can see in this video that yasippet is loaded.

![Peek 2022-10-24 21-57](https://user-images.githubusercontent.com/78803192/197607912-c2d83d57-74aa-482f-ac79-bd0204de72be.gif)

**Reply by yyoncho** [reply]
> Still the same.. can see in this video that yasippet is loaded.

and   `(yas-global-mode)` is executed? 

Can you confirm that the issue is gone after `M-x lsp-start-plain` just to make sure that it is something in your config?

**Reply by azegas** [reply]
I can confirm that after initiating `M-x lsp-start-plain` - everything works just fine..

Executing - you mean `C-x C-e` on `(yas-global-mode)` in my config? Yes, I do that, the output is `t`, but the problem in my main config still persists :/

**Reply by azegas** [reply]
my problem seems very similar to [this](https://emacs.stackexchange.com/questions/58069/rust-analyzer-completion-broken).

But as you can see in my last gif, the yas-global-mode is on.

Perhaps I missunderstand "execution"? Could you please help me with that? How could one check if (yas-global-mode) was executed correctly?

**Reply by yyoncho** [reply]
Can you upgrade all of your packages? Then restart and in your init.el delete everything and put only:

```
(add-hook 'js-mode-hook 'lsp)
(global-yas-mode 1)
``` 
(also keep the package.el related code to initialization).

For me, it is hard to guess what is causing the issue.

**Reply by yyoncho** [reply]
> (add-hook 'js-mode-hook 'lsp)

(The major mode hook might be different)

**Reply by azegas** [reply]
I think I have finally found what was causing this. Please take a look at this gif.

Seems like when i turn `company-quickhelp-mode` off - completion renders properly.

Let me know what you think.

Here is my current company [config](https://github.com/arvydasg/emacs/blob/master/recipes/company.el).

![Peek 2022-10-28 06-39](https://user-images.githubusercontent.com/78803192/198497399-2acadd2f-227f-4398-8b0c-a98ead365c15.gif)

**Reply by yyoncho** [reply]
I can confirm the issue. Feel free to turn it into a bug report. For some reason the issue happens only with ts-ls.

**Reply by yyoncho** [reply]
@arvydasg fixed in the latest lsp-mode.

**Reply by azegas** [reply]
That sounds great, @yyoncho!

Wish I could have had a chance to create my first ever bug report so I would become one of the contributer of lsp-mode :)) but you were very quick to fix it!

I have deleted my elpa folder with all the emacs packages and installed them all again upon emacs launch, but the problem persists.

Does it mean that elpa/melpa does not have the most recent lsp-mode package yet? Should it be downloaded from "source" in some way? To get the working package that you have released.

Thank you for your work!

**Reply by yyoncho** [reply]
> I have deleted my elpa folder with all the emacs packages and installed them all again upon emacs launch, but the problem persists.
> 
> Does it mean that elpa/melpa does not have the most recent lsp-mode package yet? Should it be downloaded from "source" in some way? To get the working package that you have released.

usually, it takes 1-2 hours melpa to pick the changes. It still hasn't picked the change: https://melpa.org/#/lsp-mode

**Reply by azegas** [reply]
I can confirm that now it works together with `company-quickhelp-mode`.
